enhancements/operator-framework-api/catalogue-check.md

```diff
@@ -0,0 +1,180 @@
+---
+title: neat-enhancement-idea
+authors:
+  - "@camilamacedo86"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-02-19
+last-updated: 2021-02-19
+status: implementable
+see-also:
+  - "/enhancements/olm/max-openshift-versions-for-operators.md"
+---
+
+# OpenshiftCatalogueValidator 
+
+This proposal describes a new set/suite of checks to be implemented in the [operator-framework/api][oper-api] to validate the bundle operator data regarding specific rules for the OpenShift Catalogue.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently, [operator-framework/api][oper-api] centralized the code implementation for superset/suites of checks used in the tools and across the board. See [here](https://github.com/operator-framework/api/blob/master/pkg/validation/validation.go#L38-L46) such as:
+
+- `BundleValidator` check if the operator bundle spec is accordingly to spec definition. For further information see [here](https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md).
+- `OperatorHubValidator` checks of the operator is respecting the required criteria to be published in the [OperatorHub.io](https://operatorhub.io/).
+
+The new `OpenShiftCatalogueValidator` proposed here has the purpose of perform checks which are specifically to publish the operator bundle on the OpenShift Catalogue. The validator will raise issues or warnings accordingly.
+
+This checks will also be available to be used by the users via the tools such as SDK and consequently also called in the pipeline. Note that SDK provides currently:
+
+```sh
+To validate a bundle with a superset of requirements for operatorhub.io specifically:
+   
+  $ operator-sdk bundle validate ./bundle --select-optional name=operatorhub
+``` 
+
+Then, the idea proposed here would allow the tool also provide:
+
+```sh
+To validate a bundle with a superset of requirements for OpenShift Catalogue specifically:
+   
+$ operator-sdk bundle validate ./bundle --select-optional name=openshiftcatalogue
```

Comment:
> i think we need an openshift version, because the validation rules will vary based on the openshift version (e.g. whether or not you are allowed to use v1beta1 CRDs)

If the bundle has not minKubeVersion, or maxOCPVersion or a set up in the index `com.redhat.openshift.versions` then it means that the operator bundle should work in all versions. In this way, look that we are doing these checks accordingly with the configuration of the bundle. 

- and arguably we can leave "catalog" off the name. It's validating whether the bundle is acceptable for openshift clusters.

Agreed 👍 I think it is a good suggestion.